base/stats.go

```diff
@@ -408,43 +410,37 @@ func (s *SgwIntStat) Describe(ch chan<- *prometheus.Desc) {
 }
 
 func (s *SgwIntStat) Collect(ch chan<- prometheus.Metric) {
-	ch <- prometheus.MustNewConstMetric(s.statDesc, s.statValueType, float64(s.Val), getValues(s.labels)...)
+	ch <- prometheus.MustNewConstMetric(s.statDesc, s.statValueType, float64(atomic.LoadInt64(&s.Val)), getValues(s.labels)...)
 }
 
 func (s *SgwIntStat) Set(newV int64) {
-	s.mutex.Lock()
-	defer s.mutex.Unlock()
-	s.Val = newV
+	atomic.StoreInt64(&s.Val, newV)
 }
 
 func (s *SgwIntStat) SetIfMax(newV int64) {
-	s.mutex.Lock()
-	defer s.mutex.Unlock()
-	if newV > s.Val {
-		s.Val = newV
+	for {
+		cur := atomic.LoadInt64(&s.Val)
+		nxtVal := cur + newV
+		if atomic.CompareAndSwapInt64(&s.Val, cur, nxtVal) {
+			return
+		}
 	}
 }
 
 func (s *SgwIntStat) Add(newV int64) {
-	s.mutex.Lock()
-	defer s.mutex.Unlock()
-	s.Val += newV
+	atomic.AddInt64(&s.Val, newV)
 }
 
 func (s *SgwIntStat) MarshalJSON() ([]byte, error) {
-	s.mutex.Lock()
-	defer s.mutex.Unlock()
-	return []byte(fmt.Sprintf("%v", s.Val)), nil
+	return []byte(fmt.Sprintf("%v", atomic.LoadInt64(&s.Val))), nil
```

Comment:
Using FormatInt is quicker and zero alloc, vs. Sprintf:

https://medium.com/swlh/bad-go-frivolous-sprintf-2ad28fedf1a0